recipes/openvdb/all/conanfile.py

```diff
@@ -27,91 +30,191 @@ class OpenVDBConan(ConanFile):
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["USE_BLOSC"] = self.options.with_blosc
+        tc.variables["USE_ZLIB"] = self.options.with_zlib
+        tc.variables["USE_LOG4CPLUS"] = self.options.with_log4cplus
+        tc.variables["OPENVDB_SIMD"] = self.options.simd
+        if Version(self.version) >= "10.0.0":
+            tc.variables["OPENVDB_USE_DELAYED_LOADING"] = self.options.delayed_load
+
+        tc.variables["OPENVDB_CORE_SHARED"] = self.options.shared
+        tc.variables["OPENVDB_CORE_STATIC"] = not self.options.shared
+
+        # All available options but not exposed yet. Set to default values
+        tc.variables["OPENVDB_BUILD_CORE"] = True
+        tc.variables["OPENVDB_BUILD_BINARIES"] = False
+        tc.variables["USE_EXR"] = False # This is only used for vdb_render
+        tc.variables["OPENVDB_BUILD_PYTHON_MODULE"] = False
+        tc.variables["OPENVDB_BUILD_UNITTESTS"] = False
+        tc.variables["OPENVDB_BUILD_DOCS"] = False
+        tc.variables["OPENVDB_BUILD_HOUDINI_PLUGIN"] = False
+        tc.variables["OPENVDB_BUILD_HOUDINI_ABITESTS"] = False
+
+        tc.variables["OPENVDB_BUILD_AX"] = False
+        tc.variables["OPENVDB_BUILD_AX_BINARIES"] = False
+        tc.variables["OPENVDB_BUILD_AX_UNITTESTS"] = False
+
+        tc.variables["OPENVDB_BUILD_MAYA_PLUGIN"] = False
+        tc.variables["OPENVDB_ENABLE_RPATH"] = False
+        tc.variables["OPENVDB_CXX_STRICT"] = False
+        tc.variables["USE_HOUDINI"] = False
+        tc.variables["USE_MAYA"] = False
+        tc.variables["USE_STATIC_DEPENDENCIES"] = False
+        tc.variables["USE_PKGCONFIG"] = False
+        tc.variables["OPENVDB_INSTALL_CMAKE_MODULES"] = False
+
+        if Version(self.version) >= "9.0.0":
+            tc.variables["OPENVDB_BUILD_NANOVDB"] = self.options.nanovdb
+            tc.variables["NANOVDB_USE_OPENVDB"] = self.options.nanovdb_with_openvdb
+            tc.variables["USE_TBB"] = self.options.nanovdb_with_tbb
+            tc.variables["NANOVDB_USE_INTRINSICS"] = True
+            tc.variables["NANOVDB_BUILD_TOOLS"] = False
+
+        if self._needs_boost:
+            tc.variables["Boost_USE_STATIC_LIBS"] = not self.options["boost"].shared
+            tc.variables["OPENVDB_DISABLE_BOOST_IMPLICIT_LINKING"] = True
+
+        if self._needs_openexr:
+            tc.variables["OPENEXR_USE_STATIC_LIBS"] = not self.options["openexr"].shared
+
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        # Remove FindXXX files from OpenVDB. Let Conan do the job
-        tools.remove_files_by_mask(os.path.join(self._source_subfolder, "cmake"), "Find*")
-        with open("FindBlosc.cmake", "w") as f:
+        apply_conandata_patches(self)
```

Comment:
Instead of the rather involved patching of `FindTBB.cmake`, you could perhaps simply let `find_package(TBB)` rely on the config.cmake file generated by Conan?
```suggestion
        apply_conandata_patches(self)
        rm(self, "FindTBB.cmake", os.path.join(self.source_folder, "cmake"))
        replace_in_file(self, os.path.join(self.source_folder, "openvdb", "openvdb", "CMakeLists.txt"),
                        "find_package(TBB ${MINIMUM_TBB_VERSION} REQUIRED COMPONENTS tbb)",
                        "find_package(TBB REQUIRED CONFIG)")
        replace_in_file(self, os.path.join(self.source_folder, "openvdb", "openvdb", "CMakeLists.txt"),
                        "if(OPENVDB_FUTURE_DEPRECATION AND FUTURE_MINIMUM_TBB_VERSION)",
                        "if(FALSE)")
```

Reply:
Thanks for your review @valgur ! I'll follow your other suggestions.
About this one, the patch is just a back-port from fixes done in the upstream repository. I agree with you that it's a big patch but I thought that this would be better than modifying the CMakeLists. 

I think the best solution would be that OpenVDB relied on the config files provided by its dependencies instead of maintaining its own `FindXXX` files.  There seems to be some work related in https://github.com/AcademySoftwareFoundation/openvdb/pull/1603 .  Maybe @Idclip can tell us if there is any plan to address that in the short term.